apps/api-gateway/src/geo-location/geo-location.controller.ts

```diff
@@ -0,0 +1,64 @@
+import { Controller, Get, HttpStatus, Logger, Param, Res } from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { GeoLocationService } from './geo-location.service';
+import { ApiResponseDto } from '../dtos/apiResponse.dto';
+import IResponseType from '@credebl/common/interfaces/response.interface';
+import { ResponseMessages } from '@credebl/common/response-messages';
+import { Response } from 'express';
+@Controller('/')
+@ApiTags('geolocation')
+export class GeoLocationController {
+  constructor(
+    private readonly geolocationService: GeoLocationService,
+    private readonly logger: Logger
+  ) {}
+
+  /**
+   * @returns get all countries
+   */
+  @Get('countries')
+  @ApiOperation({ summary: 'Get all countries', description: 'Get all countries' })
+  @ApiResponse({ status: HttpStatus.OK, description: 'Success', type: ApiResponseDto })
+  async getAllCountries(@Res() res: Response): Promise<Response> {
+    const countriesDetails = await this.geolocationService.getAllCountries();
+    const finalResponse: IResponseType = {
+      statusCode: HttpStatus.OK,
+      message: ResponseMessages.user.success.countriesVerificationCode,
+      data: countriesDetails
+    };
+    return res.status(HttpStatus.OK).json(finalResponse);
+  }
+
+  /**
+   * @returns get all states by countryId
+   */
+
+  @Get('countries/:countryId/states')
+  @ApiOperation({summary: 'Get all states by using countryId ', description: 'Get all states by using countryId '
+  })
+  @ApiResponse({status: HttpStatus.OK, description: 'Success', type: ApiResponseDto})
+  async getStatesByCountryId(@Param('countryId') countryId: string, @Res() res: Response): Promise<Response> {
```

Comment:
We are not using any of authentication and authentication. Is that the intended behavior or the API?